### PR TITLE
Feat: Cloud Trial Lifecycle

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
Epic Link : https://mattermost.atlassian.net/browse/MM-50719

### CODE REVIEW

File: data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-1.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+**Step 1**
+
+Ref Epic: <https://mattermost.atlassian.net/browse/MM-50719>
+
+Scenario 1: Start of trial
+
+1. Create a workspace from `https://portal.test.cloud.mattermost.com/cloud/signup`
+2. Login into the workspace
+3. Verify the Global banner is displaed with the message `Your 30-day Enterprise trial has started! Purchase before MM-DD to keep your workspace.` and `Review your options` button
```
Comment (johndavidlugtu) [inline data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-1.md:43]: Can we add to the test "Review your options" to open up the pricing modal?

File: data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-1.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+4. The Banner should be dissmissable
+
+Scenario 2: Last day of trial
+
+1. The Global banner should display a message `This is the last day of your free trail. Purchase a plan before 11.59pm PT or contact sales` and `Review your options` button
+
+Scenario 3: User accessing workspace after 30 days of trial
+
+1. Verify the message `Your trial has ended` is displayed
+2. Verify `Contact us` and `Host your own instance` buttons are shown
```
Comment (johndavidlugtu) [inline data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-1.md:53]: Can we verify "Contact us" goes to mattermost.com/contact-sales and "Host your own instance" goes to mattermost.com/download?

**Review by johndavidlugtu — CHANGES_REQUESTED** [review]

**Comment by lindalumitchell** [comment]
Just to note that I'm holding off on my review until the above changes are made (or not). :)

**Comment by yasserfaraazkhan** [comment]
@lindalumitchell 
this is up for review now 😄

File: data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-2.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+---
+# (Required) Ensure all values are filled up
+name: "Cloud Evaluation Trial | User should be able to purchase a Plan within 30 days trial period"
```
Comment (AGMETEOR) [inline data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-2.md:3]: Can only self-serve a professional plan

Comment (yasserfaraazkhan) [reply data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-2.md:3]: Oh!! I picked it from figma, it shows `Purchase` button for Enterprise 
@AGMETEOR , will enterprise card show `contact sales` in this case?

Comment (AGMETEOR) [reply data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-2.md:3]: > Oh!! I picked it from figma, it shows `Purchase` button for Enterprise @AGMETEOR , will enterprise card show `contact sales` in this case?

Yes Enterprise card only shows contact sales

File: data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-1.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+1. Create a workspace from `https://portal.test.cloud.mattermost.com/cloud/signup`
+2. Login into the workspace
+3. Verify the Global banner is displaed with the message `Your 30-day Enterprise trial has started! Purchase before MM-DD to keep your workspace.` and `Review your options` button
+4. Click on `Review your options`
+5. Verify pricing modal is opened. The modal should have `Professional` and `Enterprise` cards and have `Purchase` button on both
+6. The Banner should be dissmissable
+
+Scenario 2: Last day of trial
+
+1. The Global banner should display a message `This is the last day of your free trail. Purchase a plan before 11.59pm PT or contact sales` and `Review your options` button
```
[the remark this answers is not in the document]
Comment (johndavidlugtu) [reply data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-1.md:50]: @AGMETEOR here's the Jira ticket about that: https://mattermost.atlassian.net/browse/MM-51479

@yasserfaraazkhan minor thing: typo on "trial"

File: data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-2.md (hunk 3 above)
Comment (yasserfaraazkhan) [reply data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-2.md]: done

File: data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-1.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (yasserfaraazkhan) [reply data/test-cases/cloud/cloud-trial/cloud-evaluation-trial/test-1.md]: done

**Review by lindalumitchell — APPROVED** [review]
Thanks Yasser! LGTM 👍

**Pull request merged**